index.bs

```diff
@@ -3134,6 +3217,66 @@ and an optional [=boolean=] |isNoised| (default false):
 1. Run [=obtain and deliver an aggregatable debug report on source registration=]
     with |dataTypes|, |source|, and |isNoised|.
 
+To <dfn>find sources with common destinations and reporting origin</dfn> given an [=attribution source=] |pendingSource|:
+1. Let |matchingSources| be a new [=list=].
+1. [=set/iterate|For each=] |source| of the user agent's [=attribution source cache=]:
+    1. Let |commonDestinations| be the [=set/intersection=] of |source|'s [=attribution source/attribution destinations=] and |pendingSource|'s [=attribution source/attribution destinations=].
+    1. If |commonDestinations| [=set/is empty=], [=iteration/continue=].
+    1. If |source|'s [=attribution source/reporting origin=] and |pendingSource|'s [=attribution source/reporting origin=] are not [=same origin=], [=iteration/continue=].
+    1. [=list/Append=] |source| to |matchingSources|.
+1. Return |matchingSources|.
+
+To <dfn>check if source has incompatible attribution scope fields</dfn> given an [=attribution source=] |source| and an [=attribution source=] |pendingSource|:
+1. If |pendingSource|'s [=attribution source/attribution scope limit=] is null, return false.
+1. If |source|'s [=attribution source/attribution scope limit=] is null, return true.
+1. If |source|'s [=attribution source/max event states=] is not equal to |pendingSource|'s [=attribution source/max event states=], return true.
+1. If |source|'s [=attribution source/attribution scope limit=] is less than |pendingSource|'s [=attribution source/attribution scope limit=], return true.
+1. Return false.
+
+To <dfn>remove source and associated event reports</dfn> given an [=attribution source=] |source| and a [=moment=] |minTriggerTime|:
+1. [=set/iterate|For each=] [=event-level report=] |report| of the [=event-level report cache=]:
+    1. If |report|'s [=event-level report/source identifier=] is not equal to |source|'s [=attribution source/source identifier=], [=iteration/continue=].
+    1. If |report|'s [=event-level report/trigger time=] is less than |minTriggerTime|, [=iteration/continue=].
+    1. [=set/Remove=] |report| from the [=event-level report cache=].
+    1. [=list/Remove=] all elements from [=attribution rate-limit records=] whose [=attribution rate-limit record/entity ID=] is equal to |report|'s [=event-level report/report ID=].
```

Comment:
Do we need to check scope event versus aggregate when removing attribution rate-limit records here?

Reply:
Nice catch. Done.